dask/bytes/core.py

```diff
@@ -390,7 +390,7 @@ def get_fs_paths_myopen(urlpath, compression, mode, encoding='utf8',
                                  encoding=encoding, errors=errors, **kwargs)
         if 'w' in mode:
             paths = _expand_paths(urlpath, name_function, num)
-        elif "*" in urlpath:
+        elif "*" in urlpath or "?" in urlpath or "[" in urlpath:
```

Comment:
I also suspect that the globs in other backends are probably fragile to glob-escaped characters, not to mention that keys in s3 or GCS can maybe contain these special characters.